src/tls_stream.rs

```diff
@@ -32,7 +33,7 @@ lazy_static! {
 #[derive(Default)]
 pub struct Builder {
     domain: Option<Vec<u16>>,
-    verify_callback: Option<Arc<Fn(io::Result<()>, &CertChain) -> io::Result<()>>>,
+    verify_callback: Option<Arc<Fn(CertValidationResult) -> io::Result<()>>>,
```

Comment:
This needs to be passed a reference, not a result by value or it'll have access to a bunch of dangling pointers once the callback returns.